src/cca/scripts/background.ts

```diff
@@ -48,22 +49,59 @@ let serversPromise = GetGlobalTun2SocksVpnDevice().then((vpnDevice) => {
     network: 'Cloud',
     loginType: uproxy_core_api.LoginType.INITIAL,
   }).then(() => {
-    return new UproxyServerRepository(getLocalStorage(), core, vpnDevice);
+    try {
+      return new UproxyServerRepository(
+        getLocalStorage(), core, vpnDevice);
+    } catch (e) {
+      console.warn('local storage unavailable');
+      const serverRepository = new UproxyServerRepository({
+        length: 0,
+        clear: () => {
+          throw new Error('this is mock storage');
+        },
+        getItem: (key: string) => {
+          throw new Error('this is mock storage');
+        },
+        key: (index: number) => {
+          throw new Error('this is mock storage');
+        },
+        removeItem: (key: string) => {
+          throw new Error('this is mock storage');
+        },
+        setItem: (key: string, data: string) => {
+          throw new Error('this is mock storage');
+        }
+      }, core, vpnDevice);
+      return serverRepository;
+    }
   });
 });
 
 // Create UI.
 let serversListPagePromise: Promise<ServerListPage> = new Promise((resolve, reject) => {
   chrome.app.runtime.onLaunched.addListener(() => {
     console.debug('Chrome onLaunched fired');
-    chrome.app.window.create('index_vulcanized.html', null, (appWindow) => {
+    chrome.app.window.create('index_vulcanized.html', {
+      // When running as a Chrome extension for the first time,
+      // open in a nice Nexus 5-like 16:9 aspect ratio (vertically).
+      id: 'uProxy',
+      innerBounds: {
+        width: 400,
+        height: 712
+      }
+    }, (appWindow) => {
       console.debug('window created');
       let document = appWindow.contentWindow.document;
       document.addEventListener('DOMContentLoaded', function (event) {
         console.debug('dom ready');
         serversPromise.then((servers) => {
           console.debug('servers ready');
-          resolve(new ServerListPage(appWindow.contentWindow.document.body, servers));
+          const listElem = appWindow.contentWindow.document.body.querySelector('#server-list') as HTMLElement;
+          const serverListPage = new ServerListPage(listElem, servers);
+          appWindow.contentWindow.addEventListener('resize', () => {
```

Comment:
Correct, it will never trigger on a mobile device - but I found it useful during developing.